torch/distributed/fsdp/_optim_utils.py

```diff
@@ -858,31 +858,26 @@ def _broadcast_unsharded_pos_dim_tensor_state(
 def _rekey_sharded_optim_state_dict(
     sharded_osd: Dict[str, Any],
     model: torch.nn.Module,
-    optim_input: Optional[
-        Union[
-            List[Dict[str, Any]],
-            Iterable[torch.nn.Parameter],
-        ]
-    ] = None,
+    optim: torch.optim.Optimizer,
 ) -> Dict[str, Any]:
     """
     Rekeys the optimizer state dict from unflattened parameter names to
-    flattened parameter IDs according to the calling rank's ``optim_input``,
-    which may be different across ranks. In particular, the unflattened
-    parameter names are represented as :class:`_OptimStateKey` s.
+    flattened parameter IDs according to the calling rank's ``optim``, which
+    may be different across ranks. In particular, the unflattener parameter
```

Comment:
don't quite understand "unflattener parameter"

Reply:
```suggestion
    may be different across ranks. In particular, the unflattened parameter
```

When I first onboarded onto FSDP, I had a mental of **flattened parameters** (`FlatParameter`s) and **unflattened parameters** (original parameters), so all of the optimizer state dict work uses that language. I think now we have collectively drifted to using the term **original parameters** instead of **unflattened parameters**.

I think **original parameters** and then **FSDP parameters** might be solid language, where **FSDP parameters** generalizes (`FlatParameter`s) in case we ever choose to combine original parameters in a non-flattened way. However, I am certainly open to discussion.

Reply:
flattened/unflattened, sharded/unsharded, those are terms that usually don't mean much to external users.

Reply:
Mhm, makes sense.

I was referring to within the FSDP devs. Since we have several threads going on right now using terms like flattened/unflattened, sharded/unsharded, original/FSDP etc., it might be helpful to converge to some common terminology.